### PR TITLE
Add public API `BlockStore.Rollback`

### PR DESCRIPTION
**Pull request by yihuang**
The context: https://github.com/tendermint/tendermint/pull/8223#issuecomment-1123800668, https://github.com/tendermint/tendermint/pull/8493

So although the newest tendermint has implemented the "validate block before persist", but since the production node could be still using the old version, and when app hash mismatch happens, the pending block is already persisted, so we still want to be able to delete that pending block manually.
By exposing this public API, we can do the block store rollback in the application side.

### CODE REVIEW

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -22,4 +22,6 @@ Friendly reminder, we have a [bug bounty program](https://hackerone.com/tendermi
 
 ### IMPROVEMENTS
 
+- []() Add public API `BlockStore.Rollback`.
```
Comment (yihuang) [inline CHANGELOG_PENDING.md:25]:
```suggestion
- [#8574](https://github.com/tendermint/tendermint/pull/8574) Add public API `BlockStore.Rollback`.
```

**Comment by cmwaters** [comment]
> but since the production node could be still using the old version, and when app hash mismatch happens

We did backport these changes to v0.34. I just don't think we've cut a point release yet

**Comment by yihuang** [comment]
> > but since the production node could be still using the old version, and when app hash mismatch happens
> 
> We did backport these changes to v0.34. I just don't think we've cut a point release yet

I know, but since the current production nodes are running with old versions, and even if we release a new patch version, they don't necessarily all upgrade it, so I think it's good to support an optional `--delete-pending-block` flag in the rollback command in cosmos-sdk.

**Comment by yihuang** [comment]
> By pending block you mean a block that has been committed but hasn't been executed yet?

Yeah, more specifically, it's saved in block store, but not in the state store.

**Comment by yihuang** [comment]
don't stale

**Pull request closed without merging**